src/sagemaker/estimator.py

```diff
@@ -833,6 +875,82 @@ def __init__(self, entry_point, source_dir=None, hyperparameters=None, enable_cl
 
         self._hyperparameters = hyperparameters or {}
 
+    def _git_clone_code(self):
+        """Git clone repo containing the training scripts. This method also validate ``git_config``,
+        and set ``entry_point`` and ``source_dir`` to the right file or directory in the repo cloned.
+
+        Raises:
+            CalledProcessError: If 1. failed to clone git repo
+                                   2. failed to checkout the required branch
+                                   3. failed to checkout the required commit
+            ValueError: If 1. entry point specified does not exist in the repo
+                           2. source dir specified does not exist in the repo
+        """
+        self._validate_git_config()
+        # create a temporary directory to store the cloned repo
+        repo_dir = tempfile.mkdtemp()
+        try:
+            subprocess.check_call(['git', 'clone', self.git_config['repo'], repo_dir])
+        except subprocess.CalledProcessError:
+            raise subprocess.CalledProcessError(1, cmd='git clone {} {}'.format(self.git_config['repo'], repo_dir))
```

Comment:
Why do you need to create a new `CalledProcessError` here instead of just using the original call?